admin/src/main/scala/io/buoyant/admin/Admin.scala

```diff
@@ -1,65 +1,112 @@
+    def handle(exc: Throwable): Boolean = {
+      log.error(exc, label)
+      false
+    }
+  }
+
+  private val server = Http.server
+    .withLabel(label)
+    .withMonitor(loggingMonitor)
+    .withStatsReceiver(NullStatsReceiver)
+    .withTracer(NullTracer)
+
+  def appHandlers(app: com.twitter.app.App): Handlers = Seq(
+    "/admin/server_info" -> new TextBlockView().andThen(new ServerInfoHandler(app)),
+    "/admin/shutdown" -> new ShutdownHandler(app)
+  )
+
+  def baseHandlers: Handlers = Seq(
     "/admin/contention" -> (new TextBlockView andThen new ContentionHandler),
-    "/admin/lint" -> new LintHandler(),
-    "/admin/lint.json" -> new LintHandler(),
+    "/admin/lint" -> new LintHandler,
+    "/admin/lint.json" -> new LintHandler,
     "/admin/threads" -> new ThreadsHandler,
     "/admin/threads.json" -> new ThreadsHandler,
     "/admin/announcer" -> (new TextBlockView andThen new AnnouncerHandler),
-    "/admin/dtab" -> (new TextBlockView andThen new DtabHandler),
     "/admin/pprof/heap" -> new HeapResourceHandler,
     "/admin/pprof/profile" -> new ProfileResourceHandler(Thread.State.RUNNABLE),
     "/admin/pprof/contention" -> new ProfileResourceHandler(Thread.State.BLOCKED),
     "/admin/ping" -> new ReplyHandler("pong"),
-    "/admin/shutdown" -> new ShutdownHandler(app),
     "/admin/tracing" -> new TracingHandler,
     "/admin/logging" -> (new StyleOverrideFilter andThen new LoggingHandler),
-    "/admin/metrics" -> new MetricsQueryHandler,
-    "/admin/metrics/prometheus" -> new PrometheusStatsHandler(MetricsStatsReceiver.defaultRegistry),
-    Path.Clients -> new ClientRegistryHandler(Path.Clients),
-    Path.Servers -> new ServerRegistryHandler(Path.Servers),
-    "/admin/files/" -> ResourceHandler.fromJar(
-      baseRequestPath = "/admin/files/",
-      baseResourcePath = "twitter-server"
-    ),
     "/admin/registry.json" -> new RegistryHandler,
     "/favicon.png" -> ResourceHandler.fromJar(
       baseRequestPath = "/",
       baseResourcePath = "io/buoyant/linkerd/admin/images"
     )
-  ).map {
-      case (path, handler) =>
-        path -> (new IndexView(path, path, () => Nil) andThen handler)
-    }
+  )
 
-  private[this] def metricsRoutes: Seq[(String, Service[Request, Response])] = Seq(
+  // XXX this will be moved into telemeters soon
+  private def metricsHandlers: Seq[(String, Service[Request, Response])] = Seq(
+    "/admin/metrics" -> new MetricsQueryHandler,
+    "/admin/metrics/prometheus" -> new PrometheusStatsHandler(MetricsStatsReceiver.defaultRegistry),
     "/admin/metrics.json" -> HttpMuxer,
     "/admin/per_host_metrics.json" -> HttpMuxer
   )
 
-  def allRoutes = twitterServerRoutes ++ metricsRoutes
+  /** Generate an index of the provided handlers */
+  private def indexHandlers(handlers: Handlers): Handlers = {
+    val paths = handlers.map { case (p, _) => p }.sorted.distinct
+    val index = new IndexTxtHandler(paths)
+    Seq(
+      "/admin/index.txt" -> index,
```

Comment:
just so you have a way to list all of the admin endpoints without guessing or reading the code (a problem i personally have a lot)